pkg/service/replay/mock.go

```
@@ -288,10 +314,94 @@ func (m *mock) upload(ctx context.Context, testSetID string) error {
+	}
+
+	// After successfully writing the config, push it to the repo
+	if m.cfg.ReRecord.Branch != "" && m.cfg.ReRecord.Owner != "" {
+		err := m.pushConfigChange(ctx, testSetID, tsConfig, m.cfg.ReRecord.Owner, m.cfg.ReRecord.Branch)
+		if err != nil {
+			m.logger.Error("Failed to push config change", zap.Error(err))
+			return err
+		}
+	}
+
 	err = utils.AddToGitIgnore(m.logger, m.cfg.Path, "/*/mocks.yaml")
 	if err != nil {
 		utils.LogError(m.logger, err, "failed to add /*/mocks.yaml to .gitignore file")
 	}
 
 	return nil
 }
+
+// pushConfigChange sends a request to the api-server to push the updated config to a git branch.
+func (m *mock) pushConfigChange(ctx context.Context, testSetID string, tsConfig *models.TestSet, owner, branch string) error {
+	m.logger.Info("Attempting to push config change to git", zap.String("testSetID", testSetID), zap.String("branch", branch))
+
+	// 1. Construct the request payload
+	payload := MockChangeReq{
+		Config:    tsConfig,
+		TestSetID: testSetID,
+		Branch:    branch,
+		Owner:     owner,
+	}
+
+	body, err := json.Marshal(payload)
+	if err != nil {
+		m.logger.Error("Failed to marshal config push request payload", zap.Error(err))
+		return err
+	}
+
+	// 2. Create the HTTP request
+	url := m.cfg.APIServerURL + configPushPath
+	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		m.logger.Error("Failed to create HTTP request for config push", zap.Error(err))
+		return err
+	}
+
+	// 3. Set necessary headers
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+m.token)
+
+	// 4. Send the request
+	client := &http.Client{Timeout: 30 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		m.logger.Error("Failed to send config push request to API server", zap.Error(err))
+		return err
+	}
+	defer resp.Body.Close()
+
+	// 5. Handle the response
+	if resp.StatusCode != http.StatusOK {
+		respBody, _ := io.ReadAll(resp.Body)
```

Don't ignore the error coming from io.ReadAll.